docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,205 @@
+
+# thin-edge Data Model
+
+The **data model** identifies all data send or received from/to **thin-edge** and its components, to interact with those.
+For all data it defines format and explains behaviour.
+
+## Telemetry Data
+
+**Telemetry Data** consists of **measurements**, **events** and **alarms**. Each is defined by a set of data-elements, each with specific behaviour.
+
+### Measurements
+**measurements** carry values from physical **Sensors**[^1] or a device's **Domain Application**[^1];
+e.g. voltage and current of an electricity meter, or current state of the manufacturing control process
+
+
+A measurement can carry a **single value**, or **muliple values** all taken at a single point in time.
+```javascript
+   {
+     // example for a single-value messurement
+     "temperature":               // 'type_name' of that measurement
```

Comment:
We should rename `type_name` as we will be supporting the `type` field in the future (currently it is only added when sending measurements to Cumulocity).

Reply:
Now that "type_name" is changed to "name".